### ISSUE TITLE
Why do certain games show saved backups but others not?

### ISSUE THREAD

**Issue by darklinkpower** [mtkennerly/ludusavi-playnite#69]
Hi, just wondering about this. For a reason unknown to me, most games do show backed up items in the context menu correctly, e.g. `Persona 5 Royal`:

![image](https://github.com/mtkennerly/ludusavi-playnite/assets/1389286/bd9e24c9-16cb-403b-ba50-0f50d9fd75c8)

But others, like `Nier: Automata` do not show backed up items:

![image](https://github.com/mtkennerly/ludusavi-playnite/assets/1389286/dbc7f21e-e9e1-4992-b73d-438362c1a289)

Saves are being backed up without issues on game close:

![image](https://github.com/mtkennerly/ludusavi-playnite/assets/1389286/97536b6d-46d9-442c-a019-e13f10244f39)

Another example is `Resident Evil 2`:

![image](https://github.com/mtkennerly/ludusavi-playnite/assets/1389286/4fe19ef4-3b9d-4330-8daf-f9e3c0b28cd9)

**Comment by mtkennerly** [comment]
This is because of name mismatches like `Nier: Automata` vs `NieR: Automata`.

When performing backups/restores, the plugin uses the `ludusavi find` command to normalize names, but it doesn't do that for the backup list. I tried experimenting with using `find` when opening the context menu, but it adds a delay of about 1 second, which isn't great. Maybe I could add a long-running background task that runs `find` pre-emptively, or maybe Ludusavi should have a bulk version of `find`.

For now, if you set "look up with another title" to the normalized title, then it'll list the backups.

**Comment by darklinkpower** [comment]
I see, that makes sense.I briefly checked the source code and from what I understand Ludusavi Playnite performs a backup list check and caches it to a dictionary. As far as I know, Ludusavi manifest obtains stores Ids already that could be used to match too.

This gives me the idea of separating the backups handling to an specific class that would also support obtaining backups by id by cache'ing the storeIds
```csharp
public class BackupsManager
{
    private Dictionary<string, List<ApiBackup>> gameNameDictionary;
    private Dictionary<string, string> steamIdToNameDictionary;
    
    public void RefreshLudusaviBackups()
    {
        // Refresh gameNameDictionary
        // steamIdToNameDictionary is updated with the steam id as key and the name used in gameNameDictionary as a value
    }

    public List<ApiBackup> GetBackups(string gameName)
    {
        // Do what is already being done
    }

    public List<ApiBackup> GetBackupsBySteamId(string steamId)
    {
        if (steamIdToNameDictionary.TryGetValue(steamId, out var gameName))
        {
            return GetBackups(gameName);
        }

        return somethingsomething;
    }

    public string GetGameBackups(Game game)
    {
        if (gameNameDictionary.ContainsKey(storeId))
        {
            return gameNameDictionary[storeId];
        }
        else if (game.PluginId == steamPluginId)
        {
            return GetBackupsBySteamId(game.Id)
        }
    }

    ....
}
```

In this implementation, `steamIdToNameDictionary` contains the store Id as a key, and their values will be the key name to be used in `gameNameDictionary` to obtain the real backups. Other stores could be handled the same way.

In my opinion this approach could work, while making it more extensible in the future but without knowing how Ludusavi itself works I don't know if it's feasible. Separating backup concerns from the main plugin class is also in my opinion an added benefit.

**Comment by mtkennerly** [comment]
This will be fixed once Ludusavi 0.24.0 comes out. It will include an `api` command that supports bulk title queries.
